src/views/Alteration.vue

```diff
@@ -107,29 +118,33 @@ export default class Alteration extends Mixins(LegalApiMixin, FilingTemplateMixi
         priority: false
       })
 
-      if (this.alterationId) {
-        // store the filing ID
-        this.setFilingId(this.alterationId)
-
-        // fetch draft alteration to resume
-        const alterationFiling = await this.fetchFilingById(this.alterationId)
-
-        // do not proceed if this isn't an ALTERATION filing
-        if (!alterationFiling.alteration) {
-          throw new Error('Invalid Alteration filing')
-        }
-
-        // do not proceed if this isn't a DRAFT filing
-        if (alterationFiling.header.status !== FilingStatus.DRAFT) {
-          throw new Error('Invalid Alteration status')
-        }
-
-        // parse alteration filing into store
-        this.parseAlteration(alterationFiling)
-      } else {
-        // as we don't have the necessary query params, do not proceed
-        throw new Error('Invalid alteration filing ID')
-      }
+      // TODO: Handle Returning from a DRAFT alteration filing
+      // if (this.alterationId) {
+      // store the filing ID
+      // this.setFilingId(this.alterationId)
+
+      // fetch draft alteration to resume
+      // const alterationFiling = await this.fetchFilingById(this.alterationId)
+
+      // // do not proceed if this isn't an ALTERATION filing
+      // if (!alterationFiling.alteration) {
+      //   throw new Error('Invalid Alteration filing')
+      // }
+      //
+      // // do not proceed if this isn't a DRAFT filing
+      // if (alterationFiling.header.status !== FilingStatus.DRAFT) {
+      //   throw new Error('Invalid Alteration status')
+      // }
+      //
+      // // parse alteration filing into store
+      // this.parseAlteration(alterationFiling)
+      // } else {
+      //   // as we don't have the necessary query params, do not proceed
+      //   throw new Error('Invalid alteration filing ID')
+      // }
+
+      const businessSnapshot = await this.fetchBusinessSnapshot()
+      await this.parseBusinessSnapshot(businessSnapshot)
```

Comment:
You should handle the "promise rejected" state (ie, catch).

Reply:
^^ would happen in case of fetch error

Your "get" methods could return an exception, which becomes a rejected promise when awaiting. (Or you could wrap everything in a try-catch block, but you should still handle it somehow.)

Reply:
Wouldn't the thrown errors be caught by the parent Try/Catch block and emit from line 151?

Reply:
Oops, I didn't notice you were already in a try-catch block.

Yes, yes they would.